ci/do_ci.sh

```diff
@@ -167,6 +183,41 @@ elif [[ "$1" == "check_format" ]]; then
 elif [[ "$1" == "docs" ]]; then
   docs/publish.sh
   exit 0
+elif [[ "$1" == "github_release" ]]; then
+  if [[ ! -f "${ENVOY_SRCDIR}/build_release_stripped/envoy" ]]; then
+      echo "could not locate envoy binary at path: ${ENVOY_SRCDIR}/build_release_stripped/envoy"
+      
+      # TODO(taion809): discuss whether or not failing to publish to github warrents failing the build itself
```

Comment:
I think misconfiguration that is this bad, i.e. we're in a completely unexpected environment, should warrant failure via `exit 1`.